src/scout_apm/core/tracked_request.py

```diff
@@ -103,7 +104,8 @@ def finish(self):
         if self.is_real_request():
             self.tag("mem_delta", Memory.get_delta(self.memory_start))
             if not self.is_ignored():
-                RequestManager.instance().add_request(self)
+                batch_command = BatchCommand.from_tracked_request(self)
+                AgentContext.socket().send(batch_command)
```

Comment:
Right, missed copying that over!

Reply:
Although `BatchCommand.from_tracked_request` has a couple defensive pathways to return `None`, I don't think they are actually possible:

* The first condition is that all spans in `complete_spans` have a non-`None` `end_time`, but this is always the case given that spans go through `stop()` before being added to `complete_spans`
* The second condition is that the request has a non-`None` `end_time`, but this is checked and set just a couple lines above in `TrackedRequest.finish()`

Also this is the only call site, and there are many other fields within tracked request/span that if not correct would lead to a bad message being sent to the core agent.

Therefore I've removed the defensive pathways.